nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -274,107 +271,103 @@ define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'u
+                this.updateModelParameterValue(message);
+            });
+
+            this.paramsWidget = ParamsWidget.make({
+                bus: this.paramsBus,
+                workspaceId: this.runtime.workspaceId(),
+                paramIds: this.internalModel.getItem(['inputs', 'otherParamIds']),
+                initialParams: this.internalModel.getItem(['params']),
+                initialDisplay: this.internalModel.getItem(['inputs', 'paramDisplay']) || {},
+            });
+
+            return this.paramsWidget.start({
+                node,
+                parameters: this.spec.getSpec().parameters,
+            });
+        }
+
+        startInputWidgets() {
+            this.paramsNode = document.createElement('div');
+            return Promise.try(() => {
+                this.buildParamsWidget(this.paramsNode).then(() => {
+                    this.updateAppConfigState();
+                });
+            });
+        }
+
+        stopInputWidgets() {
+            if (this.paramsWidget) {
+                return Promise.try(() => {
+                    this.paramsWidget.stop();
+                });
+            }
+            return Promise.resolve();
+        }
+
+        /**
+         * Updates the stored parameter value using the bus message passed by the input widget.
+         *
+         * This is followed by evaluating the state of all inputs to see if the form is valid
+         * and ready for submission.
+         *
+         * @param {object} message - the bus message returned from the widget with a changed
+         *  parameter value
+         */
+        updateModelParameterValue(message) {
+            this.internalModel.setItem(['params', message.parameter], message.newValue);
+            this.internalModel.setItem(
+                ['inputs', 'paramDisplay', message.parameter],
+                message.newDisplayValue
+            );
+
+            if (message.isError) {
+                return;
+            }
+
+            return this.updateAppConfigState();
+        }
+
+        /**
+         * Validates the current form values, updates the state of the form to either 'complete'
+         * or 'incomplete', and sets the form submission button to disabled if the form is incomplete.
+         */
+
+        async updateAppConfigState() {
+            const paramIds = this.internalModel.getItem(['inputs', 'otherParamIds']),
+                paramValues = this.internalModel.getItem(['params']);
+
+            const results = await this.spec.validateParams(paramIds, paramValues, {});
+            const isValid = Object.values(results).every((param) => param.isValid);
+            this.internalModel.setItem(['state', 'params'], isValid ? 'complete' : 'incomplete');
+            const confirmButton = $(document.querySelector('#' + this.id)).nearest(
+                '[data-element="ok"]'
+            );
+            confirmButton[0].disabled = !isValid;
```

Comment:
all important part: enable or disable the modal's confirm button, depending on whether or not the form config is valid